cpp/src/arrow/compute/kernels/hash.cc

```diff
@@ -110,15 +166,19 @@ class DictEncodeAction : public ActionBase {
 
   Status Reserve(const int64_t length) { return indices_builder_.Reserve(length); }
 
-  void ObserveNull() { indices_builder_.UnsafeAppendNull(); }
+  Status ObserveNull() {
+    indices_builder_.UnsafeAppendNull();
+    return Status::OK();
+  }
 
   template <class Index>
-  void ObserveFound(Index index) {
+  Status ObserveFound(Index index) {
     indices_builder_.UnsafeAppend(index);
+    return Status::OK();
```

Comment:
I wonder if this has performance implications. I'll run the perf benchmarks before/after to see